src/core/server/workspaces/workspaces_service.ts

```diff
@@ -43,11 +43,29 @@ export class WorkspacesService
     this.logger = coreContext.logger.get('workspaces-service');
   }
 
+  private proxyWorkspaceTrafficToRealHandler(setupDeps: WorkspacesSetupDeps) {
+    /**
+     * Proxy all {basePath}/w/{workspaceId}{osdPath*} paths to
+     * {basePath}{osdPath*}
+     */
+    setupDeps.http.registerOnPreRouting((request, response, toolkit) => {
```

Comment:
There is an `registerOnPreRouting` in `http_server`, will it conflict with this?